Build failure: ax25-apps

## Steps To Reproduce

Steps to reproduce the behavior:

1. build ax25-apps

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
gcc -DHAVE_CONFIG_H -I. -I..  -DAX25_SYSCONFDIR=\""/etc/ax25"\" -DAX25_LOCALSTATEDIR=\""/var/lib/ax25"\"   -g -O2 -Wall -c -o routing.o routing.c
routing.c: In function 'route_add':
routing.c:83:13: error: implicit declaration of function 'malloc' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
   83 |             malloc(sizeof(struct route_table_entry));
      |             ^~~~~~
routing.c:17:1: note: include '<stdlib.h>' or provide a declaration of 'malloc'
   16 | #include "ax25ipd.h"
  +++ |+#include <stdlib.h>
```

</details>

## Additional context

Might be another gcc 14 one

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.7, NixOS, 25.05 (Warbler), 25.05beta732261.6df24922a140`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

## Notify maintainers

@sarcasticadmin

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ax/ax25-apps/package.nix
+++ b/pkgs/by-name/ax/ax25-apps/package.nix
@@ -1,25 +1,29 @@
 {
   lib,
   stdenv,
-  fetchurl,
+  fetchFromGitHub,
+  autoreconfHook,
   libax25,
   ncurses,
 }:
 
 stdenv.mkDerivation rec {
   pname = "ax25-apps";
-  version = "0.0.8-rc5";
+  version = "0.0.8-rc5-unstable-2021-05-13";
 
   buildInputs = [
+    autoreconfHook
     libax25
     ncurses
   ];
 
-  # Due to recent unsolvable administrative domain problems with linux-ax25.org,
-  # the new domain is linux-ax25.in-berlin.de
-  src = fetchurl {
-    url = "https://linux-ax25.in-berlin.de/pub/ax25-apps/ax25-apps-${version}.tar.gz";
-    sha256 = "sha256-MzQOIyy5tbJKmojMrgtOcsaQTFJvs3rqt2hUgholz5Y=";
+  # src from linux-ax25.in-berlin.de remote has been
+  # unreliable, pointing to github mirror from the radiocatalog
+  src = fetchFromGitHub {
+    owner = "radiocatalog";
+    repo = "ax25-apps";
+    rev = "afc4a5faa01a24c4da1d152b901066405f36adb6";
+    hash = "sha256-RLeFndis2OhIkJPLD+YfEUrJdZL33huVzlHq+kGq7dA=";
   };
 
   configureFlags = [

--- a/pkgs/by-name/ax/ax25-apps/package.nix
+++ b/pkgs/by-name/ax/ax25-apps/package.nix
@@ -7,7 +7,7 @@
   ncurses,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
   pname = "ax25-apps";
   version = "0.0.8-rc5-unstable-2021-05-13";
 
@@ -32,11 +32,11 @@ stdenv.mkDerivation rec {
     "--program-transform-name=s@^call$@ax&@;s@^listen$@ax&@"
   ];
 
-  meta = with lib; {
+  meta = {
     description = "AX.25 ham radio applications";
     homepage = "https://linux-ax25.in-berlin.de/wiki/Main_Page";
-    license = licenses.lgpl21Only;
-    maintainers = with maintainers; [ sarcasticadmin ];
-    platforms = platforms.linux;
+    license = lib.licenses.lgpl21Only;
+    maintainers = with lib.maintainers; [ sarcasticadmin ];
+    platforms = lib.platforms.linux;
   };
 }

